### ISSUE TITLE
Rates and Transit Times API

### ISSUE THREAD

**Issue by danielmcguinness** [WhatArmy/FedexRest#17]
Hi guys. So I forked your repo and thought I would contribute by adding support for the rates API. 

I've tried to stay within the same code structure as you are using for the Ship API, and was hoping to get a draft PR open this week. 

However I've noticed an odd issue that I cannot reconcile with the documentation, hopefully you can help me to understand.

I'm following this documentation https://developer.fedex.com/api/en-us/catalog/rate/v1/docs.html and it works fine when I send specific parameters. For example, if I dont send rateRequestType (not required according to the docs) as part of the requestedShipment, it fails. If I send serviceType or shipDateStamp (both optional) as part of the requestedShipment, it fails.

Am I following out of date documentation? The error code is always "REQUEST.MISMATCH".

Seems that it will accept ONLY required parameters WITH rateRequestType. Any deviation from that causes a failure.

Thank you!

**Comment by greggles** [comment]
Do you have a technical contact at FedEx? I'd suggest asking them for advice.

If you don't have one I can try reaching out to the person who helped us a year ago. I'd want a copy of request/response if you could provide that.

**Comment by danielmcguinness** [comment]
@greggles I believe its because the Rates API is virtualized so I have to end exactly whats contained in the JSON API Collection. Any deviation causes a failure.

**Comment by danielmcguinness** [comment]
@greggles I'm pretty confident now that its due to the virtualized response. I've taken examples from the JSON Collection and they all work and do not have additional `alerts` regarding information having been changed. (Just the one which states "This is a Virtual Response.").

Thank you for your offer to reach out to your contact though, very much appreciated!

**Comment by code2prog** [comment]
@danielmcguinness They did it strangely with this virtualization. Recently I had to change tests because the virtualized sandbox started returning strange things.
The old sandbox for that had stability problems. Tracking API tests once passed and once ended with a 500 error on the Fedex side xD

**Comment by danielmcguinness** [comment]
@code2prog Agreed, I saw the same when testing locally. The behaviour is documented [here](https://developer.fedex.com/api/en-us/guides/sandboxvirtualization.html)
